fix(Rating): fix issue with state snapshots and therefore provide decent support for mobile phones

LisaKirchhofer opened the pull request:
Corresponding issue can be found here: https://github.com/rsuite/rsuite/issues/2917

The problem was the following: 2 functions operating on the same state were wrapped within 1 parent function -> one of them setting a state, the other one reading the state. The reading function was expecting to get the new value from the state, set by the previous function. As react operates on snapshots, the reading function retrieved the old value instead of the new one. I took the opportunity and tried to refactor those parts.

Commit:
fix(Rating): fix issue with state snapshots and therefore provide decent support for mobile phones

Commit:
fix(Rating): remove unused return statement

LisaKirchhofer commented:
@simonguo or @SevenOutman can you please have a look? :)

simonguo commented:
hi @LisaKirchhofer  
You need to add tests for the fixed code.  The test script can be run by the following command:
```
M=Rate npm run tdd
```

LisaKirchhofer commented:
Hi @simonguo!
I did not change or extend any of the functionality. All of it is already covered in the existing code - I implemented against them. So the question is, which kind of Test do you expect?
The bug I fixed was a react specific one, rather than a functional one.
Thanks in advance, Lisa

simonguo commented:
The issues in #2917 are currently not covered by our test scripts. If this PR solves #2917, I hope there will be a corresponding test case. In the test script, the click and keyboard can be simulated, and the github action will also automatically run the test script in Chrome and Firefox browsers.
